src/tools/sign-oci-layer-root-hashes/src/registry.rs

```diff
@@ -406,19 +408,94 @@ fn decompress_file(compressed_path: &Path, decompressed_path: &Path) -> Result<(
     Ok(())
 }
 
-fn attach_tarfs_index(path: &Path) -> Result<()> {
-    info!("Adding tarfs index to layer");
-    let mut file = std::fs::OpenOptions::new()
-        .read(true)
+// fn attach_tarfs_index(path: &Path) -> Result<()> {
+//     info!("Adding tarfs index to layer");
+//     let mut file = std::fs::OpenOptions::new()
+//         .read(true)
+//         .write(true)
+//         .open(path)
+//         .context("Failed to open {path} for writing")?;
+//     tarindex::append_index(&mut file)
+//         .map_err(|e| anyhow!(e))
+//         .context("Failed to append index")?;
+//     file.flush()
+//         .map_err(|e| anyhow!(e))
+//         .context("Failed to flush file changes")?;
+
+//     Ok(())
+// }
+
+fn attach_erofs_meta(path: &Path) -> Result<()> {
+    info!("Creating erofs meta. Appending decompressed tar to erofs meta");
+
+    // Create an erofs image using mkfs.erofs
+    let erofs_path = path.with_extension("erofs");
+    debug!(
+        "Creating erofs meta image {:?} from {:?}",
+        &erofs_path, &path
+    );
+
+    // TODO: need to double check how we are using -T and -U
+    let status = Command::new("mkfs.erofs")
+        .args([
+            "--tar=i",
+            "-T", "0", // zero out unix time
```

Comment:
Hi! I think you could append `--mkfs-time` together with `-T 0` here to only clear out the mkfs time in the superblock, but keep per-inode `mtime`.

Reply:
Just confirming, using "-T 0" defaults to "-T 0 --all-time" right? (Link - https://github.com/erofs/erofs-utils/blob/dev/mkfs/main.c#L159)

I tested using "-T 0 --mkfs-time" and created two erofs metdata and they had the same sha256sum, and hence they were reproducible. Just to confirm, if we use "-T 0 --mkfs-time", the erofs metadata will still be reproducible right?

In addition, is there any issue with not capturing the per-inode mtime? Trying to understand this better, appreciate your advice/help!